src/processes.jl

```diff
@@ -23,16 +23,16 @@ Generate `n` realizations of spatial point process `p`
 inside geometry `g`. Optionally specify sampling
 algorithm `algo` and random number generator `rng`.
 """
-Base.rand(rng::Random.AbstractRNG, p::PointProcess, g::Geometry, n::Int; algo=default_sampling_algorithm(p, g)) =
+Base.rand(rng::Random.AbstractRNG, p::PointProcess, g::GeometryOrMesh, n::Int; algo=default_sampling_algorithm(p, g)) =
```

Comment:
Let's omit the type annotation in this case and leave `g` without type. That way we avoid introducing the union type and everything should work fine in the Meshes.jl side with proper types.

```suggestion
Base.rand(rng::Random.AbstractRNG, p::PointProcess, g, n::Int; algo=default_sampling_algorithm(p, g)) =
```